src/AWSS3.jl

```diff
@@ -195,19 +195,36 @@ end
 
 s3_get_meta(a...; b...) = s3_get_meta(global_aws_config(), a...; b...)
 
+function _s3_exists_file(aws::AbstractAWSConfig, bucket, path; kw...)
+    q = Dict("prefix" => path, "delimiter" => "", "max-keys" => 1)
+    l = S3.list_objects_v2(bucket, q; aws_config=aws)
+    c = get(l, "Contents", nothing)
+    isnothing(c) && return false
+    return get(c, "Key", "") == path
+end
+
+function _s3_exists_dir(aws::AbstractAWSConfig, bucket, path; kw...)
+    a = string(path)[1:(end - 1)] * "."
```

Comment:
safer in case someone tries to make a unicode s3 path for some reason:
```suggestion
    a = chop(string(path))* "."
```

Reply:
Good catch, I always think in the back of my mind that string indexing is by character...